docs/Getting Started.md

```diff
@@ -0,0 +1,778 @@
+
+![Application Pipeline](images/pipe.001.jpeg?raw=true "Application Pipeline")
+
+One of the important features of cloudflow architecture is the complete separation of the components from how they are connected as part of the pipeline. The _streamlets_ described above are the individual building blocks of the pipeline. You can connect them using a declarative language that forms the _blueprint_ of the pipeline. Streamlets can be shared across blueprints making them individual reusable objects. And just like streamlets, blueprints also form an independent component of a cloudflow application.
+
+### Project structure
+
+Here's how we would structure a typical cloudflow application as a project.
+
+```
+   |-project
+   |-src
+   |---main
+   |-----avro
+   |-----blueprint
+   |-----resources
+   |-----scala
+   |-------sensordata
+   |-build.sbt
+```
+
+This is a Scala project and we have the following structural components at the leaf level of the above tree:
+
+* **avro** : contains the avro schema of the domain objects
+* **blueprint** : contains the blueprint of the application in a file named `blueprint.conf`
+* **scala** : contains the source code of the application under the package name `sensordata`
+* **build.sbt** : the sbt build script
+
+### The sbt build script
+
+Here's a minimal version of the sbt build script:
+
+```
+import sbt._
+import sbt.Keys._
+
+lazy val sensorData =  (project in file("."))
+  .enablePlugins(CloudflowAkkaStreamsApplicationPlugin)
+  .settings(
+    libraryDependencies ++= Seq(
+      "com.lightbend.akka"     %% "akka-stream-alpakka-file"  % "1.1.2",
+      "com.typesafe.akka"      %% "akka-http-spray-json"      % "10.1.10",
+      "ch.qos.logback"         %  "logback-classic"           % "1.2.3",
+      "com.typesafe.akka"      %% "akka-http-testkit"         % "10.1.10" % "test"
+    ),
+    name := "sensor-data",
+    organization := "com.lightbend",
+
+    scalaVersion := "2.12.10",
+    crossScalaVersions := Vector(scalaVersion.value)
+  )
+```
+
+cloudflow offers a few plugins that abstract quite a bit of boilerplates necessary to build a complete application. In this example we use the plugin `CloudflowAkkaStreamsApplicationPlugin` that provides you all building blocks of developing an Akka Streams based cloudflow application.
+
+> **Note:** You can use multiple plugins to develop an application that uses multiple runtimes (Akka, Spark, Flink etc.). For simplicity of this example we will be using only one.
+ 
+The above build script is standard Scala sbt - the only difference is the lugin which we provide as part of cloudflow.
+
+### Schema first approach
+
+In cloudflow streamlets work with optional inputs and outputs that are statically typed. The types represent objects that the specific input / output can handle. The first step in application development is to encode the objects in the form of avro schema. cloudflow will generate appropriate classes corresponding to each schema.
```

Comment:
Let's point to some avro resource link for the user who is not familiar enough. Also we should point to the avro sbt plugin we use right?

Reply:
done ..